### PR TITLE
[chore]: enable negative-positive rule from testifylint

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

Testifylint is a linter that provides best practices with the use of testify.

This PR enables [negative-positive](https://github.com/Antonboom/testifylint?tab=readme-ov-file#negative-positive) rule from [testifylint](https://github.com/Antonboom/testifylint)

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

File: internal/otelarrow/test/e2e_test.go [hunk 1]
```diff
@@ -358,13 +358,13 @@ func failureMemoryLimitEnding(t *testing.T, _ testParams, testCon *testConsumer,
 	rSigs, rMsgs := logSigs(testCon.recvLogs)
 
 	// Test for arrow stream errors.
-	require.Less(t, 0, eSigs["arrow stream error|||code///message///where"], "should have exporter arrow stream errors: %v", eSigs)
-	require.Less(t, 0, rSigs["arrow stream error|||code///message///where"], "should have receiver arrow stream errors: %v", rSigs)
+	require.Positive(t, eSigs["arrow stream error|||code///message///where"], "should have exporter arrow stream errors: %v", eSigs)
```
Comment (codeboten) [inline internal/otelarrow/test/e2e_test.go:361]: Should these be Negative?

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go [hunk 2]
```diff
@@ -110,8 +110,8 @@ func TestScrape(t *testing.T) {
 				var scraperErr scrapererror.PartialScrapeError
 				require.ErrorAs(t, err, &scraperErr)
 				noProcessesErrored := scraperErr.Failed
-				require.Lessf(t, 0, noProcessesErrored, "Failed to scrape metrics - : error, but 0 failed process %v", err)
-				require.Lessf(t, 0, noProcessesScraped, "Failed to scrape metrics - : 0 successful scrapes %v", err)
+				require.Positivef(t, noProcessesErrored, "Failed to scrape metrics - : error, but 0 failed process %v", err)
```
Comment (codeboten) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:113]: Same question here

File: receiver/influxdbreceiver/receiver_test.go [hunk 3]
```diff
@@ -77,7 +77,7 @@ func TestWriteLineProtocol_v2API(t *testing.T) {
 		require.NoError(t, err)
 
 		metrics := nextConsumer.lastMetricsConsumed
-		if assert.NotNil(t, metrics) && assert.Less(t, 0, metrics.DataPointCount()) {
+		if assert.NotNil(t, metrics) && assert.Positive(t, metrics.DataPointCount()) {
```
Comment (codeboten) [inline receiver/influxdbreceiver/receiver_test.go:80]: Here as well

File: internal/otelarrow/test/e2e_test.go (hunk 1 above)
Comment (codeboten) [reply internal/otelarrow/test/e2e_test.go]: 🤦🏻 right.... thanks, i missed this detail

**Review by codeboten — APPROVED** [review]

**Pull request merged**